modules/akka/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/akka/ShardingAggregate.scala

```diff
@@ -21,12 +22,12 @@ import scala.util.control.NonFatal
 final class ShardingAggregate[Evt: Typeable, St: Typeable, Cmd, Rej: Typeable](
     override val name: String,
     ref: ActorRef,
-    pq: CurrentEventsByPersistenceIdQuery,
+    journalPluginId: String,
     logger: LoggingAdapter
-)(implicit ec: ExecutionContext, mt: Materializer, tm: Timeout)
+)(implicit as: ActorSystem, ec: ExecutionContext, mt: Materializer, tm: Timeout)
```

Comment:
If you pass the `ActorSystem` here, why do you need the `ExecutionContext ` and the `Materializer`? Both can be derived form the `ActorSystem`, right?